packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/DeviceMotionModule.kt

```diff
@@ -0,0 +1,230 @@
+package expo.modules.sensors.modules
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener2
+import android.hardware.SensorManager
+import android.os.Bundle
+import android.view.Choreographer
+import android.view.Surface
+import android.view.WindowManager
+import expo.modules.interfaces.sensors.SensorServiceInterface
+import expo.modules.interfaces.sensors.SensorServiceSubscriptionInterface
+import expo.modules.interfaces.sensors.services.AccelerometerServiceInterface
+import expo.modules.interfaces.sensors.services.GravitySensorServiceInterface
+import expo.modules.interfaces.sensors.services.GyroscopeServiceInterface
+import expo.modules.interfaces.sensors.services.LinearAccelerationSensorServiceInterface
+import expo.modules.interfaces.sensors.services.RotationVectorSensorServiceInterface
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ExpoMethod
+import org.unimodules.core.interfaces.services.EventEmitter
+import org.unimodules.core.interfaces.services.UIManager
+
+class DeviceMotionModule(context: Context?) : ExportedModule(context), SensorEventListener2 {
+  private var mLastUpdate: Long = 0
+  private var mUpdateInterval = 1.0f / 60.0f
+  private val mRotationMatrix = FloatArray(9)
+  private val mRotationResult = FloatArray(3)
+  private var mAccelerationEvent: SensorEvent? = null
+  private var mAccelerationIncludingGravityEvent: SensorEvent? = null
+  private var mRotationEvent: SensorEvent? = null
+  private var mRotationRateEvent: SensorEvent? = null
+  private var mGravityEvent: SensorEvent? = null
+  private lateinit var mServiceSubscriptions: MutableList<SensorServiceSubscriptionInterface>
+  private lateinit var mUiManager: UIManager
```

Comment:
We tend not to decapitalize abbreviations like this one 😉 
```suggestion
  private lateinit var mUIManager: UIManager
```